coralproject/elkhorn
# [A11y] - Accessibility in Ask Forms

# Accessibility
### Things I've learned and Resources

An amazing course by [Marcy Sutton](http://twitter.com/marcysutton) 

> https://egghead.io/courses/start-building-accessible-web-applications-today

The usage of labelledby

> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

Relying on the title attribute

> https://www.paciellogroup.com/blog/2012/01/html5-accessibility-chops-title-attribute-use-and-abuse/

A guide to use good practices to make your sites accessible

> http://webaim.org/techniques/forms/controls

Another great article for building forms

> https://24ways.org/2015/the-accessibility-mindset/

How to structure forms

> https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Forms/How_to_structure_an_HTML_form

W3C Specification on A11y

> https://www.w3.org/TR/WCAG20/ 
> Tutorials from the W3C to make accessble sites
> https://www.w3.org/WAI/tutorials/forms/
> https://www.w3.org/WAI/intro/wcag.php
> https://www.w3.org/WAI/tutorials/forms/labels/

Tools to test your site

> http://wave.webaim.org/report#/https://coralproject.net/contact.html

Who to follow to know more about a11y

> Talita Pagani [@talitapagani](http://twitter.com/talitapagani) 
> Marcy Sutton [@marcysutton](http://twitter.com/marcysutton) 
## Whats new with this PR?
- Accesibility added in Ask Forms
## How to test this PR?
- Run Cay and Elkhorn
- Create a new Form
- Add all the fields you can
- Check the source code and look the for attr and id usage with labels
- Use a Screen Reader or fire VoiceOver in Mac 

Note: I couldn't test the standalone form due to a auth issue 

------------------------------ EDIT: 

Dealing with this Chrome issue lead my to find a lot of how a user interacts with the accesibility tools. 

I found this article http://webaim.org/techniques/keyboard/ that tells you how a user uses the keyboard when he's navigating a Website.
Basically, the way to navigate Radio Butons is with `<Arrows Keys>` or pressing `<Control> + <Option> + <Arrow Key>`

<img width="793" alt="screen shot 2016-10-19 at 12 18 11 pm" src="https://cloud.githubusercontent.com/assets/1401559/19525218/9bd6b770-95f6-11e6-95bf-2e43817e5b88.png">

To test: 
- Go to this form that I just created with this branch - https://s3.amazonaws.com/nyt-stage/580004390cefe54d86e2c731.html
- Turn on VoiceOver `<Command> + <f5>`
- Navigate with `<Tab>` 
- When you get to the Radiobuttons you will get a message like this

> <img width="1078" alt="screen shot 2016-10-19 at 12 27 18 pm" src="https://cloud.githubusercontent.com/assets/1401559/19525476/6ef9a46e-95f7-11e6-8f75-6f32c3adbc1c.png">
- Wait 3 seconds and this message should appear: (People that already use the software already know this, if not you wait and you get this hint)

> <img width="779" alt="screen shot 2016-10-19 at 12 22 57 pm" src="https://cloud.githubusercontent.com/assets/1401559/19525440/4fa9155e-95f7-11e6-9e9c-c7b61531ee04.png">
- Navigate using `<Arrows Keys>` or pressing `<Control> + <Option> + <Arrow Key>`
# DEMO TIME :D!

![a11yoptions](https://cloud.githubusercontent.com/assets/1401559/19525714/4687977e-95f8-11e6-9b81-8122a9876bf7.gif)
